src/sales/dto/sale.dto.ts

```diff
@@ -0,0 +1,9 @@
+export class CreateSaleDTO {
+    readonly subtotal: number;
+    readonly total: number;
+    readonly cartItem:[{
+        productId:string;
+        quantity:number;
+    }];
+    readonly createdAt: Date;
+}
```

Comment:
Crea el dto para el cart item y válida los datos con los decoradores de class-validator

```suggestion
import { IsNotEmpty, IsMongoId, IsNumber, IsArray } from 'class-validator';
import { Types } from 'mongoose';

export class CreateCartItemDTO {
  @IsNotEmpty()
  @IsMongoId()
  readonly product: Types.ObjectId;

  @IsNotEmpty()
  @IsNumber()
  readonly quantity: number;
}

export class CreateSaleDTO {
    @IsNotEmpty()
    @IsNumber()
    readonly subtotal: number;
    
    @IsNotEmpty()
    @IsNumber()
    readonly total: number;
    
    @IsNotEmpty()
    @IsArray()
    readonly cart:  CreateCartItemDTO[]
}
```